src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/PropertyPages/BuildPropertyPage.xaml

```diff
@@ -175,37 +175,48 @@
 
   <EnumProperty Name="WarningLevel"
                 DisplayName="Warning level"
-                Description="Specifies the level to display for compiler warnings."
+                Description="Specifies the level to display for compiler warnings. Higher levels produce more warnings, and include all warnings from lower levels."
                 HelpUrl="https://go.microsoft.com/fwlink/?linkid=2146798"
                 Category="ErrorsAndWarnings">
     <EnumValue Name="0"
-               DisplayName="0" />
+               DisplayName="0 - Disable all warnings" />
     <EnumValue Name="1"
-               DisplayName="1" />
+               DisplayName="1 - Severe warning messages" />
     <EnumValue Name="2"
-               DisplayName="2" />
+               DisplayName="2 - Less severe warnings, such as warnings about hiding class members" />
     <EnumValue Name="3"
-               DisplayName="3" />
+               DisplayName="3 - Less severe warnings, such as warnings about expressions that always evaluate to true or false" />
     <EnumValue Name="4"
-               DisplayName="4" />
+               DisplayName="4 - Informational warnings" />
     <EnumValue Name="5"
-               DisplayName="5" />
+               DisplayName="5 - Warnings from C# 9" />
   </EnumProperty>
 
   <StringProperty Name="NoWarn"
                   DisplayName="Suppress specific warnings"
                   Description="Blocks the compiler from generating the specified warnings. Separate multiple warning numbers with a comma (',') or semicolon (';')."
                   HelpUrl="https://go.microsoft.com/fwlink/?linkid=2147300"
-                  Category="ErrorsAndWarnings" />
+                  Category="ErrorsAndWarnings">
+    <StringProperty.Metadata>
+      <NameValuePair Name="VisibilityCondition">
+        <NameValuePair.Value>(not (has-evaluated-value "Build" "WarningLevel" "0"))</NameValuePair.Value>
```

Comment:
Now that I think about it, I don't think we can condition `NoWarn`, `TreatWarningsAsErrors`, `WarningsAsErrors`, or `WarningsNotAsErrors` on the value of `WarningLevel`. Analyzers may report warnings regardless of the `WarningLevel`, and these other properties can control which of those are promoted to errors or suppressed.

Reply:
Good point. Fixed. Maybe we should explain this in the description of the WarningLevel property somehow. Can you think of a nice addition to the current text:

> Specifies the level to display for compiler warnings. Higher levels produce more warnings, and include all warnings from lower levels.

Reply:
In talking with the compiler team about #7977, the expectation is that, moving forward, the `WarningLevel` will primarily be about compiler warnings _and_ warnings from analyzers built into the SDK. 3rd party analyzers are not generally expected to use `WarningLevel`. So I think I would write something like:

> Specifies the level to display for compiler warnings and .NET SDK analyzer warnings. Higher levels produce more warnings, and include all warnings from lower levels.